src/test/regress/expected/multi_truncate.out

```diff
@@ -364,5 +374,122 @@ SELECT * FROM test_local_truncate;
 
+DROP TABLE test_local_truncate;
+-- Test truncate_local_table UDF
+CREATE TABLE referenced_table(id int UNIQUE, test_column int);
+INSERT INTO referenced_table SELECT x,x FROM generate_series(1,10000) x;
+CREATE TABLE referencing_table(id int, ref_id int REFERENCES referenced_table(id));
+INSERT INTO referencing_table SELECT * FROM referenced_table;
+-- Test foreign keys from local tables to distributed tables
+-- We can not truncate local tables until all the local foreign keys are removed.
+SELECT create_distributed_table('referenced_table', 'id');
+NOTICE:  Copying data from local table...
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- The following will fail, as local referencing_table has fk references
+SELECT truncate_local_table('referenced_table');
+ERROR:  cannot truncate a table referenced in a foreign key constraint
+DETAIL:  Table "referencing_table" references "referenced_table".
+HINT:  Truncate table "referencing_table" at the same time, or use TRUNCATE ... CASCADE.
+-- Test foreign keys between distributed table
+-- The following will fail as there are local records with foreign keys to this table
+SELECT create_distributed_table('referencing_table', 'ref_id');
+NOTICE:  Copying data from local table...
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT truncate_local_table('referenced_table');
+ERROR:  cannot truncate a table referenced in a foreign key constraint
+DETAIL:  Table "referencing_table" references "referenced_table".
+HINT:  Truncate table "referencing_table" at the same time, or use TRUNCATE ... CASCADE.
+-- To truncate the tables, first truncate the table that has foreign key references
+-- then the referenced table
+BEGIN;
+SELECT truncate_local_table('referencing_table');
+ truncate_local_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT * FROM table_sizes;
+       name        |  size
+---------------------------------------------------------------------
+ referenced_table  | 384 kB
+ referencing_table | 0 bytes
+(2 rows)
+
+-- I expect the following to pass, but it fails
+SELECT truncate_local_table('referenced_table');
+ERROR:  cannot truncate a table referenced in a foreign key constraint
+DETAIL:  Table "referencing_table" references "referenced_table".
+HINT:  Truncate table "referencing_table" at the same time, or use TRUNCATE ... CASCADE.
+ROLLBACK;
+-- Let's try to otruncate the tables outside a transaction
+SELECT truncate_local_table('referencing_table');
+ truncate_local_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT * FROM table_sizes;
+       name        |  size
+---------------------------------------------------------------------
+ referenced_table  | 384 kB
+ referencing_table | 0 bytes
+(2 rows)
+
+-- Similarly I expect the following to pass, but it fails
```

Comment:
Need to figure out why this does not work

Reply:
See similar comments on https://github.com/citusdata/citus/pull/3752#discussion_r408017895